frontend/src/components/ui/BuyMenu.jsx

```diff
@@ -215,6 +221,31 @@ export default function BuyMenu({game}) {
         )
     });
 
+    const undoBuild = () => {
+        if (!undoStack.length) return;
+        deselectAll();
+        const {x, y, z, label, price, tileType} = undoStack.at(-1);
+        
+        switch (label) {
+            case 'BuildTower':
+                game.removeTower(x, y);
+                gameMap.removeTower(x, y);
```

Comment:
Hmmm, not sure how I feel about having the towers in two places.

Reply:
That's been in place for a while. I agree, but it will need to be its own issue.